huaweicloud/services/iotda/resource_huaweicloud_iotda_batchtask_file.go

```diff
@@ -0,0 +1,145 @@
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/huaweicloud-sdk-go-v3/core/def"
+	"github.com/huaweicloud/huaweicloud-sdk-go-v3/services/iotda/v5/model"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+// @API IoTDA POST /v5/iot/{project_id}/batchtask-files
+// @API IoTDA GET /v5/iot/{project_id}/batchtask-files
+// @API IoTDA DELETE /v5/iot/{project_id}/batchtask-files/{file_id}
+func ResourceBatchTaskFile() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceBatchTaskFileCreate,
+		ReadContext:   resourceBatchTaskFileRead,
+		DeleteContext: resourceBatchTaskFileDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"content": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceBatchTaskFileCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.HcIoTdaV5Client(region)
+	if err != nil {
+		return diag.Errorf("error creating IoTDA v5 client: %s", err)
+	}
+
+	file, err := os.Open(d.Get("content").(string))
+	if err != nil {
+		return diag.Errorf("error opening batch task file: %s", err)
+	}
+	defer file.Close()
+
+	body := model.UploadBatchTaskFileRequestBody{
+		File: &def.FilePart{
+			Content: file,
+		},
+	}
+	createOpt := model.UploadBatchTaskFileRequest{
+		Body: &body,
+	}
+	resp, err := client.UploadBatchTaskFile(&createOpt)
+	if err != nil {
+		return diag.Errorf("error uploading IoTDA batch task file: %s", err)
+	}
+
+	if resp == nil || resp.FileId == nil {
+		return diag.Errorf("error uploading IoTDA batch task file: ID is not found in API response")
+	}
+
+	d.SetId(*resp.FileId)
+
+	return resourceBatchTaskFileRead(ctx, d, meta)
+}
+
+func resourceBatchTaskFileRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.HcIoTdaV5Client(region)
+	if err != nil {
+		return diag.Errorf("error creating IoTDA v5 client: %s", err)
+	}
+
+	resp, err := client.ListBatchTaskFiles(&model.ListBatchTaskFilesRequest{})
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error querying IoTDA batch task files")
+	}
+
+	file := &model.BatchTaskFile{}
+	for _, respFile := range *resp.Files {
+		if *respFile.FileId == d.Id() {
+			copiedFile := respFile
+			file = &copiedFile
+			break
+		}
+	}
```

Comment:
you can try to use this style:
```suggestion
	fileID := d.Id()
	allFiles := *resp.Files
	targetFile := new(model.BatchTaskFile)
	for i := range allFiles {
		if allFiles[i].FileId == fileID {
			targetFile = allFiles[i]
			break
		}
	}
```

Reply:
fixed.